pgjdbc/src/main/java/org/postgresql/core/Utils.java

```diff
@@ -110,8 +87,7 @@ private static void doAppendEscapedLiteral(Appendable sbuf, String value,
     try {
       if (standardConformingStrings) {
         // With standard_conforming_strings on, escape only single-quotes.
-        for (int i = 0; i < value.length(); ++i) {
-          char ch = value.charAt(i);
+        for (char ch : value.toCharArray()) {
```

Comment:
I wonder if this change would result array allocation and extra cpu/memory consumption issues